### ISSUE TITLE
Add a send/confirmation button

### ISSUE THREAD

**Issue by diemelone** [nextcloud/polls#1574]
**Is your feature request related to a problem? Please describe.**
Users always send mails if their answers were stored which is quite annoying

**Describe the goal you'd like to achieve**
A visual confirmation for users that their answers were send successful

**Describe alternatives you've considered**
none

**Describe possible solutions**
a button that says "send" or "ok"

**Additional context**
After clicking the send button
![grafik](https://user-images.githubusercontent.com/46229652/116242019-cd219e00-a765-11eb-9344-c144cfb4d654.png)

Users (in this case Admin) will see their own results like this
![grafik](https://user-images.githubusercontent.com/46229652/116242246-0528e100-a766-11eb-8284-261060f65923.png)

But still have the option to edit their answers when clicking the option in the three dot menu

**Comment by dartcafe** [comment]
Please take a look at #1519 and the related PR #1556. 

Since all vote clicks are stored instantly a send button would signal another behavior of the app.

**Comment by diemelone** [comment]
I have seen #1556 already, #1519 was new but thats exactly what is needed for users who are used to  threema polls, doodle, dudle, etc.
Instant storing is great, dont get me wrong but a button that gives users the expected visual feedback is crucial.

I maybe should change the request as followed:

- Add a button that says "finished editing"
- If a user clicks on this button the checkboxes are replaced with the green/red/yellow rectangles and the "finished editing" button switches to "edit vote"

In this case you do not suggest that a vote is not saved, you just enable the expected visual feedback that the users vote shows up as any other vote.
Since closing the tab doesn't delete the votes you might want to consider a better visible delete button.

before clicking "finish editing"
![grafik](https://user-images.githubusercontent.com/46229652/116359903-a44de700-a7ff-11eb-804b-25850f74bedb.png)


after clicking "finish editing
![grafik](https://user-images.githubusercontent.com/46229652/116360135-e414ce80-a7ff-11eb-9d9b-0b27d4c8eacd.png)

**Comment by dartcafe** [comment]
But in the end, we just add two buttons with no function than enable and disable editng the votes. And even if the finish vote button is clicked, there is no real function, because the votes are saved anyway.

**Comment by dartcafe** [comment]
Ok, correction: One button. Deleting the votes is a function wich is reserved to the poll owner, ATM.
Current implementation: 
![grafik](https://user-images.githubusercontent.com/26707476/116434936-27e0f580-a84b-11eb-88c3-7eee34e347ba.png)

**Comment by diemelone** [comment]
Exaclty, a button that doesn't do more than changing css classes which gives users a visual confirmation.
Is a delete button not possible or do you chose not to enable one? Because If it is possible it should be an option for a user to delete his/her vote entirely dont you think so?

**Comment by dartcafe** [comment]
> Exaclty, a button that doesn't do more than changing css classes which gives users a visual confirmation.

I am not sure, what you mean with visual confirmation, but I would say it doesn't confirm anything but the fact, that the user clicked the finish button. 

More than this, the button could pretend to have a function like, "do not count my votes until I finished."

IMHO just changing the visual state has no value compared to the costs of the delelopment, testing and maintenance.

> Is a delete button not possible or do you chose not to enable one? 

It is just not implemented ATM.

> Because If it is possible it should be an option for a user to delete his/her vote entirely dont you think so?

Yes, makes sense. Do you want to open an issue for that?

**Comment by diemelone** [comment]
Well I think users would value a feature like that quite highly, you as a developer of course understand the way these polls work entirely because you designed them. But users, esp. users that visit the site for the first time seem very uncertain whether their vote was submitted or not. This issue would give that visual confirmation that users are used to and expect for poll apps like this one.

I am no developer so I can not really estimate the effort it needs but in my understanding this would be a few lines of js which only deletes active (and currentUser) from the css classes
- `vote-item yes active currentUser` to `vote-item yes`
- `vote-item maybe active currentUser` to `vote-item maybe`
- `vote-item active (no) currentUser` to `vote-item (no)`

This should work i guess https://stackoverflow.com/a/49162726/11001992

> IMHO just changing the visual state has no value compared to the costs of the delelopment, testing and maintenance.

Do you really think this feature would need any maintenance at all? If you don't change the way these polls work with the css classes, nothing really would change, would it? And if it is, just put it to to the experimental features, if something there doesn't work properly everyone will understand.

Anyways, I'd really appreciate a feature like that but if you find it unnecessary I won't bother to ask anymore.

> Do you really think this feature would need any maintenance at all? If you don't change the way these polls work with the css classes, nothing really would change, would it?

Done #1578

**Comment by dartcafe** [comment]
I think, I understand your request and your suggested solution, but I just think it is no good idea to offer an action button, which pretends to trigger an action, which it doesn't. 

> I am not sure, what you mean with visual confirmation, but I would say it doesn't confirm anything but the fact, that the user clicked the finish button.
> 
> More than this, the button could pretend to have a function like, "do not count my votes until I finished."

Just changing the visual state of the poll options is no function. The visual feedback will i.e. signal a temporary state, where the votes were not counted until i "close" my voting. 

And yes, every line of code causes maintenance and rises the complexity, some more, some less. This code would not be the only one, which manipulates the visual state (i.e. poll status, allowed option per user or allowed users per option). And at least we have to add functions for the initial state, possibly depending on other attributes. We have to check, if the poll is open or not and hide the button. Closing and reopening can cause additionally checks and tests. 

At least we added a visual confirmation, that the vote is saved in #1556 by using a commonly used notification.

**Comment by mMuck** [comment]
IMHO with the #1556 toast there is a valid/common visual user feedback and the possibility to leave the instant update like it is. So no need for changes. Could be closed?
Am I right, this isn't published yet? This toast is very neccessary. Had some trouble with users don't understanding that they already voted leading other users to wrong conclusions... 🙁

**Comment by dartcafe** [comment]
> Am I right, this isn't published yet?

You're right. Think, we will publish it in the comming weeks. Some tests and fixes are necessary.

**Comment by mMuck** [comment]
👍 Thanks a lot.
IMO this can be closed

**Comment by dartcafe** [comment]
Since we added a visual feedback and a submit button will have no function, I close this.

**Comment by ridhwn** [comment]
Hi may i know this will be implement because it is so weird most of the end users dont really understand they have submitted the poll or not. 

I understand your technical reason but putting a submit button below the pool and just show a page saying your vote has been submitted shouldn't be a big issue. The current visual feedback is too small at the top right and people wont notice them. Can you consider? even the skype or teams pool once we choose answer it will show different page.

**Comment by dartcafe** [comment]
> I understand your technical reason but putting a submit button below the pool and just show a page saying your vote has been submitted shouldn't be a big issue.

And as long as they do not push the button, they think the votes are not saved. But they are. What is more confusing?

**Comment by ridhwn** [comment]
Yes simple as it is.

You talking about how people think that is confusing or you in confuse on how people think? this should be fairly basic understanding of UX. 

Like here in github i have to press Comment button for me to publish my comment. Even tough github developer can make without the button. 

Not sure what is the problem here where everyone else coming here telling you should have that. The toast notification is so small, people just dont see it.

**Comment by dartcafe** [comment]
### Ok. Another try with examples, supposed we have the submit button: 

A user voted for 4 options. After each click he sees the toast message, that the votes are registered. So he knows and understands, that the clicks are saved and counted. An additional button, whichs says `Submit` looks strange. "What is this button for? The app already said, the votes are counted." He clicks, a message says `Votes submitted`. Then you have two messages, stating the same information but after different actions and one of them is fake. He may ask: "When are the votes saved, right after vote click, as the toast message said or after the button click?" --> Confusion --> Uncertainty --> bad UX. 😕 

Another user voted for 4 options. After each click he misses the toast notification and sees the button and wants to submit his votes. He clicks, a message says `Votes submitted`. --> User is happy. 🎉 

A third user voted for 4 options. After each click he misses the toast notification and sees the button and wants to submit his votes. But then he decides not to vote and closes the page. Later he reopens the poll and is confused, because there are votes in his name, he didn't submit.  --> Confusion --> Uncertainty --> bad UX. 😖 

### If we stay with the current solution: 
A forth user voted for 4 options. After each click he misses the toast notification and then wonders, where he can submit his vote. Now he is trapped. What could he do? 
Option A: He refreshes the page. -> Votes are there ... Wondering ... "Ah. They are saved instantly" --> Not optimal UX, but Lesson learned 🎉  
Option B: He is desperated, closes the poll and complains the poll owner. He explains: "Votes are saved instantly". User is not happy: "I did not know that, but now I know. But I still don't like this." He returns to the page: his votes are saved. --> Not optimal UX, but Lesson learned 😐 

### Long story short
I predict, the suggested solution produces more problems, than it solves. 

> Like here in github i have to press Comment button for me to publish my comment. 

Of course you can compare a multiline editing comment with a checkbox, but you shouldn't. 😉 

> Not sure what is the problem here where everyone else coming here telling you should have that.

Everyone, hmmm. 😑
